lib/index.js

```diff
@@ -57,13 +56,7 @@ class WebpackManifestPlugin {
       hook.tap(hookOptions, normalModuleLoader);
     });
 
-    if (webpack.version.startsWith('4')) {
-      compiler.hooks.emit.tap(hookOptions, emit);
-    } else {
-      compiler.hooks.thisCompilation.tap(hookOptions, (compilation) => {
-        compilation.hooks.processAssets.tap(hookOptions, () => emit(compilation));
-      });
-    }
+    compiler.hooks.emit.tap(hookOptions, emit);
```

Comment:
so we put this different hook in for v5 based on direction from the webpack team itself. I'm actually surprised that the tests are passing with this change. I'll have to investigate recent changes in v5.x to see if something negated the need for the different hooks.

Reply:
I see, I think you are referring to https://github.com/webpack/webpack/issues/11425 and the `DeprecationWarning: Compilation.assets will be frozen in future, all modifications are deprecated.
BREAKING CHANGE: No more changes should happen to Compilation.assets after sealing the Compilation.` deprecation warning. It seems there are still a lot of plugins that are still using the emit hook and thus are called after `webpack-manifest-plugin` is done. So I think your code is the way to go but other plugins aren't ready yet.

Reply:
Meanwhile I brought this incompatibility issue to the attention to the attention of the authors of the plugins from that I know of cause an issue with your updated code. See: https://github.com/WordPress/gutenberg/issues/27984#issuecomment-762100067 and https://github.com/roots/palette-webpack-plugin/issues/13.

So you are doing it right for sure but since you have done that before a lot of plugins are updated this now causes issues probably a lot of not yet updated plugins still using that old `emit` hook. 

Maybe it is possible to make something like a check function at the end of the compilation that gives a warning if other plugins are adding assets too late and thus are not added to the manifest?

Reply:
I gave this one some serious thought. I see where the issues for your builds are valid and aren't going away anytime soon - yet another symptom of adopting new webpack majors within the first year they're available. There's never been a smooth webpack major release, they're always a mess.

I think the right thing to do here is to introduce a new option: `useLegacyEmit`, which will use the webpack v4 emit pattern even for webpack v5. It should be `false` by default, and the documentation for that option should state explain the scenario that you're facing and why the option is needed, and should also warn users that the `DeprecationWarning` will be visible. That should maintain compatibility with the folks relying on this plugin to emit in the new patterns, as well as allow folks to work around that.

Reply:
Thank you for your reply. I do think this is a very good middle ground for now! Maybe it's a good idea to write a standard text that users can utilize to make an issue at other plugin repositories to point authors at said issue and a way to fix it in their webpack plugins. I am willing to write a first version if you'd like to.